runtime/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1383,19 +1383,19 @@ private int decodePublishV5(
                 publishLimit = publish.limit();
             }
 
-            final MqttPublishHeader mqttPublishHeader = mqttPublishHeaderRO.reset();
+            final MqttPublishHeader mqttPublishHeader = new MqttPublishHeader();
```

Comment:
This does not seem to be following the same coding pattern we use for other server decoders.

I think this `MqttPublishHeader` is being kept to store information about the currently decoded `PUBLISH` frame so that it can be used later on in the decode.

However, the approach we usually take is to have `decodedXXX` or `decodeableXXX` fields describing fields either already observed (e.g. topic, qos, etc) or anticipated (e.g. remaining bytes). This can work because we decode one packet at a time, so even a large `PUBLISH` packet is decoded fully before moving on to the next packet.

Then these fields are used directly or indirectly as needed to pass parameters to `onDecodeXXX` methods.

Please let me know if there's any reason why we might need to take a different approach here for MQTT decode.